### PR TITLE
Flucca/feat/1015/add overview path

### LINKED ISSUE #1015: Add `overviewPath` to asset docs schema
**Summary**

Design spec: https://www.figma.com/file/Cs94zXOUUEizXNdDz263OA/Component-demo?node-id=114%3A169975

Because the Storybook embed will be achieved through a new MDX component, we need a file to use that new component.

We can add an `overviewPath` to `asset.docs` in the schema to render content immediately below the dashboard. See: https://github.com/carbon-design-system/carbon-platform/blob/main/docs/resource-schemas.md#asset-docs

**Acceptance criteria**

What requirements must be met to complete this request?

- [x] Add `overviewPath`
- [x] Render that markdown content immediately below the dashboard in the asset overview page template

(We might need to create a new `Contributing > Assets` page to house that and other documentation templates.)

### PR DESCRIPTION
**Pull request by francinelucca**
Closes #1015

Design: https://www.figma.com/file/Cs94zXOUUEizXNdDz263OA/Component-demo?node-id=114%3A169975
Adds overviewPath to schema, uses it in asset detail to render content below dashboard; also validates url to make sure it doesn't leave the current repository

#### Changelog

**New**

- overviewPath section in asset detail page
- overviewPath docs + schema
- `getGithubTree` function
- asset docs inheritance and defaults (note using asset's response path to construct overviewPath when relative)
- H2s as pagenav in asset detail overview tab
- AssetDetails component

**Changed**

- handle absolute urls from `getMdxSource`
- refactor asset index page to reduce file size and make more comprehensive

#### Testing / reviewing

Based on mock lib: https://github.com/francinelucca/library-testing/blob/main/overview-path/carbon.yml

http://localhost:3000/libraries/overview-path-library/latest/assets/accordion: should have overview content 2 under the dashboard
http://localhost:3000/libraries/overview-path-library/latest/assets/accordion-2: should not have overview content under the dashboard (invalid url)
http://localhost:3000/libraries/overview-path-library/latest/assets/accordion-3: should have overview content under the dashboard (not specified so uses default)
http://localhost:3000/libraries/overview-path-library/latest/assets/accordion-4: should have `overview content testinggg` under the dashboard (not specified so uses default but relative to it's own carbon.yml)

*** All items with overview content should have the h2s as a navigable item in pagenav

TODO:
(after review)

- [x] Remove mockedData in services/web-app/data/libraries.mjs

### CODE REVIEW

**Review by alisonjoseph — APPROVED** [review]

**Comment by francinelucca** [comment]
@alisonjoseph re-requested review 'cause I changed a bunch of things upon discovering some edge cass 🙏

**Review by alisonjoseph — APPROVED** [review]
Re-reviewed and everything seems to be rendering correctly

**Comment by francinelucca** [comment]
@alisonjoseph just need you to test the pageNav

**Review by alisonjoseph — APPROVED** [review]
Looks good!

File: services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/assets/[asset]/index.js [hunk 1]
```diff
@@ -85,10 +119,10 @@ const Asset = ({ libraryData, params }) => {
   const { maintainer } = assetData.params
   const MaintainerIcon = teams[maintainer] ? teams[maintainer].pictogram : Svg64Community
 
-  const isPathAbsolute = (path) => {
+  const isPathAbsolute = (urlPath) => {
```
Comment (jdharvey-ibm) [inline services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/assets/[asset]/index.js:122]: Could the `createUrl` utility function you'd made be used here instead? This regex only catches *some* cases of absolute URLs, but does not catch things like `//example.com/some-route`

**Review by jharvey10 — COMMENTED** [review]
I found it a bit challenging to review the `services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/assets/[asset]/index.js` file. It's almost 500 lines long and doesn't have any documentation.

Are there any opportunities in that file to break it into smaller modules?

That file would also greatly benefit from documentation throughout.

File: services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/assets/[asset]/index.js (hunk 1 above)
Comment (francinelucca) [reply services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/assets/[asset]/index.js]: Done!

**Comment by francinelucca** [comment]
> I found it a bit challenging to review the `services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/assets/[asset]/index.js` file. It's almost 500 lines long and doesn't have any documentation.
> 
> Are there any opportunities in that file to break it into smaller modules?
> 
> That file would also greatly benefit from documentation throughout.

@jdharvey-ibm  Abstracted most of the jsx into an `AssetDetails` component and refactore/cleaned up some of the logic, also offloaded some common functions between the asset index and the asset tabs pages into utils for reuse. What were you looking for in terms of documentation? 🤔

File: services/web-app/lib/github.js [hunk 2]
```diff
@@ -490,6 +490,46 @@ export const getLibraryRelatedLibs = async (libData) => {
   return relatedLibs
 }
 
+/**
+ * Compiles an object list of all known tags
```
Comment (jdharvey-ibm) [inline services/web-app/lib/github.js:494]: Is this "tags" as in framework tags?

**Comment by jdharvey-ibm** [comment]
@francinelucca awesome! I'll call out a couple places I think could benefit from docs in the code as review comments.

File: services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/assets/[asset]/index.js [hunk 3]
```diff
@@ -331,9 +178,40 @@ const Asset = ({ libraryData, params }) => {
...
+      message: PropTypes.string,
+      stack: PropTypes.string,
+      position: PropTypes.string
+    }),
+    warnings: PropTypes.arrayOf(PropTypes.string)
+  }),
   params: paramsPropTypes
 }
 
+const getOverviewMdxSource = async (assetData, libraryData) => {
```
Comment (jdharvey-ibm) [inline services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/assets/[asset]/index.js:199]: This function could benefit from some JSDoc or inline comments.

File: services/web-app/utils/mdx.js [hunk 4]
```diff
@@ -94,3 +95,24 @@ export async function processMdxSource(mdxSource, url) {
     warnings
   }
 }
+
+export async function getProcessedMdxSource(params, url) {
```
Comment (jdharvey-ibm) [inline services/web-app/utils/mdx.js:99]: This function could benefit from some JSDoc explaining what it does, why you'd use it, and what happens for both the happy path and the error paths.

**Review by jharvey10 — CHANGES_REQUESTED** [review]

File: services/web-app/lib/github.js (hunk 2 above)
Comment (francinelucca) [reply services/web-app/lib/github.js:494]: hahaah copy/paste got me, I need to update this description. this function just compiles an array of other assets related to a given asset (other frameworks)

File: services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/assets/[asset]/index.js (hunk 3 above)
Comment (francinelucca) [reply services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/assets/[asset]/index.js:206]: Done!

File: services/web-app/utils/mdx.js (hunk 4 above)
Comment (francinelucca) [reply services/web-app/utils/mdx.js:107]: Done!

File: services/web-app/lib/github.js (hunk 2 above)
Comment (francinelucca) [reply services/web-app/lib/github.js]: Done!

File: services/web-app/components/asset-details/asset-details.module.scss [hunk 5]
```diff
@@ -0,0 +1,26 @@
+@use '@carbon/react/scss/breakpoint' as breakpoint;
```
[the remark this answers is not in the document]
Comment (francinelucca) [reply services/web-app/components/asset-details/asset-details.module.scss:8]: Done! is there a way we can get the missing copyright notice also on scss files? 👀

[the remark this answers is not in the document]
Comment (jdharvey-ibm) [reply services/web-app/components/asset-details/asset-details.module.scss:8]: Depends on if you want to pair up and do some AST stuff 😂 

I have a TODO to add a couple stylelint rules for us, but we're going to have to make them custom.

**Review by jharvey10 — APPROVED** [review]

File: services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/assets/[asset]/index.js [hunk 6]
```diff
@@ -196,6 +196,13 @@ Asset.propTypes = {
   params: paramsPropTypes
 }
 
+/**
+ * Given an asset, retrieves the overview section mdx source content if path defined in asset index
+ * @param {import('../../../../../../../../../typedefs').Asset} assetData
```
[the remark this answers is not in the document]
Comment (francinelucca) [reply services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/assets/[asset]/index.js:201]: HAHA I'll try that

[the remark this answers is not in the document]
Comment (francinelucca) [reply services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/assets/[asset]/index.js]: Done ✨

**Pull request merged**